cpp/src/parquet/statistics_test.cc

```diff
@@ -645,6 +652,50 @@ class TestStatisticsHasFlag : public TestStatistics<TestType> {
                              EXPECT_FALSE(merged_statistics->HasDistinctCount());
                              EXPECT_FALSE(merged_statistics->Encode().has_distinct_count);
                            });
+
+    // Create a statistics object with zero distinct count. Merging preserves the distinct
+    // count if either side is zero.
+    std::shared_ptr<TypedStatistics<TestType>> statistics3;
+    std::shared_ptr<TypedStatistics<TestType>> statistics4;
+    {
+      EncodedStatistics encoded_statistics3;
+      encoded_statistics3.has_distinct_count = true;
+      encoded_statistics3.distinct_count = 0;
+      statistics3 = std::dynamic_pointer_cast<TypedStatistics<TestType>>(
+          Statistics::Make(this->schema_.Column(0), &encoded_statistics3,
+                           /*num_values=*/0));
+      EXPECT_TRUE(statistics3->HasDistinctCount());
+
+      EncodedStatistics encoded_statistics4;
+      encoded_statistics4.has_distinct_count = true;
+      encoded_statistics4.distinct_count = 10;
+      statistics4 = std::dynamic_pointer_cast<TypedStatistics<TestType>>(
+          Statistics::Make(this->schema_.Column(0), &encoded_statistics4,
+                           /*num_values=*/10));
+      EXPECT_TRUE(statistics4->HasDistinctCount());
+    }
```

Comment:
There starts to be a lot of copy-pasted code here. How about defining specific helpers? For example:
```c++
std::optional<int64_t> MergeDistinctCount(std::optional<int64_t> initial, std::vector<std::optional<int64_t>> subsequent)
```

And then
```c++
ASSERT_EQ(std::nullopt, MergeDistinctCount(std::nullopt, {0});
ASSERT_EQ(std::nullopt, MergeDistinctCount(std::nullopt, {10, 0});
ASSERT_EQ({10}, MergeDistinctCount({10}, {0, 0});
ASSERT_EQ({10}, MergeDistinctCount({0}, {10, 0});
ASSERT_EQ({10}, MergeDistinctCount({0}, {0, 10});
ASSERT_EQ(std::nullopt, MergeDistinctCount({10}, {0, 10});
ASSERT_EQ(std::nullopt, MergeDistinctCount({10}, {0, std::nullopt});
// etc?
```

Reply:
Added a `MergeDistinctCount` test helper; PTAL.